src/main/bash/sdkman-utils.sh

```diff
@@ -30,20 +30,42 @@ function __sdkman_secure_curl {
 	fi
 }
 
-function __sdkman_secure_curl_download {
+function __sdkman_secure_download {
     local curl_params="--progress-bar --location"
+    local aria2_params="--min-split-size=1M --split=10 --max-connection-per-server=10 --summary-interval=0"
 	if [[ "${sdkman_insecure_ssl}" == 'true' ]]; then
 		curl_params="$curl_params --insecure"
+		aria2_params="$aria2_params --check-certificate=false"
 	fi
 
 	local cookie_file="${SDKMAN_DIR}/var/cookie"
 
 	if [[ -f "$cookie_file" ]]; then
 	    local cookie=$(cat "$cookie_file")
 		curl_params="$curl_params --cookie $cookie"
+		aria2_params="$aria2_params --load-cookies=$cookie_file"
 	fi
 
-	curl ${curl_params} "$1"
+	if [[ -z "${sdkman_download_tool}" ]] && which aria2c &>/dev/null; then
+		sdkman_download_tool='aria2'
+		echo sdkman_download_tool=aria2 >> "${SDKMAN_DIR}/etc/config"
+		echo "Find aria2c in PATH. Choose aria2 for downloading automatically."
+		echo "If you want to use curl, set 'sdkman_download_tool=curl' in '${SDKMAN_DIR}/etc/config'"
```

Comment:
Really? Should we be forcing users into using aria2c like this?

I also don't like the verbosity of the output as well as the side-effect of writing config introduced here. This config should be written by the [selfupdate.sh](https://github.com/sdkman/sdkman-candidates-legacy/blob/master/src/main/scripts/selfupdate.sh) script in the legacy candidates service. I also think the default configuration should be `curl`, and we should update the [website's config section](https://github.com/sdkman/sdkman-website/blob/master/site/src/site/pages/usage/config.groovy) to explain the new configuration.